numpy/_core/tests/test_stringdtype.py

```diff
@@ -1324,6 +1336,46 @@ def test_strip_ljust_rjust_consistency(string_array, unicode_array):
     )
 
 
+def test_unset_na_coercion():
+    # a dtype instance with an unset na object is compatible
+    # with a dtype that has one set
+
+    # this tests uses the "add" ufunc but all ufuncs that accept more
```

Comment:
The test actually checks comparisons too, but doesn't really matter - just if for some other reason it is worth pushing another version.